src/process_utility.c

```diff
@@ -609,25 +611,58 @@ verify_constraint_list(List *constraint_list)
 
 	foreach(lc, constraint_list)
 	{
-		Constraint *constraint = (Constraint *) lfirst(lc);
+		Constraint *constraint = lfirst(lc);
 
 		verify_constraint(constraint);
 	}
 }
 
-/* process all create table commands to make sure their constraints are kosher */
+/*
+ * Process create table statements.
+ *
+ * For regular tables, we need to ensure that they don't have any foreign key
+ * constraints that point to hypertables.
+ *
+ * NOTE that this function should be called after parse analysis (in an end DDL
+ * trigger or by running parse analysis manually).
+ */
 static void
-process_create_table(Node *parsetree)
+process_create_table_end(Node *parsetree)
 {
 	CreateStmt *stmt = (CreateStmt *) parsetree;
 	ListCell   *lc;
 
 	verify_constraint_list(stmt->constraints);
+
+	/*
+	 * Only after parse analyis does tableElts contain only ColumnDefs. So, if
+	 * we capture this in processUtility, we should be prepared to have
+	 * constraint nodes and TableLikeClauses intermixed
+	 */
 	foreach(lc, stmt->tableElts)
 	{
-		ColumnDef  *column_def = (ColumnDef *) lfirst(lc);
+		ColumnDef  *coldef;
+
+		switch (nodeTag(lfirst(lc)))
+		{
+			case T_ColumnDef:
+				coldef = lfirst(lc);
+				verify_constraint_list(coldef->constraints);
+				break;
+			case T_Constraint:
 
-		verify_constraint_list(column_def->constraints);
+				/*
+				 * There should be no Constraints in the list after parse
+				 * analysis, but this case is included anyway for completeness
+				 */
+				verify_constraint(lfirst(lc));
+				break;
+			case T_TableLikeClause:
+				/* Some as above case */
```

Comment:
should there be a `verify_constraint` here? In general I am more in favor of erroring in these two cases if the are unexpected.